WSTEAMA-346: Migrate ad container to src/app/components

karinathomasbbc opened the pull request:
Resolves JIRA https://jira.dev.bbc.co.uk/browse/WSTEAMA-346

First attempt: https://github.com/bbc/simorgh/pull/10761 

Overall changes
======
Moves the Ad component from the legacy location to src/app/components, and upgrade to TypeScript

Code changes
======
- Migrate component
- Update all references
- Fix any broken tests

Testing
======

karinathomasbbc committed:
Move Ad component to src/app/components

karinathomasbbc committed:
Move Direction to Global types so that it can be reused elsewhere

karinathomasbbc committed:
Adding margins / gutters to ThemeProvider spacings

karinathomasbbc committed:
Rename & migrate styles.const

karinathomasbbc committed:
WIP Migrate Ads

karinathomasbbc committed:
Migrate canonical component

karinathomasbbc committed:
Export group3 & 4 breakpoints

karinathomasbbc committed:
Update className is optional

karinathomasbbc committed:
Migrate AdSlot

karinathomasbbc committed:
More type defs for amp

karinathomasbbc committed:
Update test

karinathomasbbc committed:
Update Amp styles

karinathomasbbc committed:
New styling for AMP Ad

karinathomasbbc committed:
Update snapshots

karinathomasbbc committed:
Suppress error messages

karinathomasbbc committed:
Delete styles file which is no longer required

karinathomasbbc commented on src/app/components/Ad/Canonical/MPU/index.tsx:
This is a rename / move from [src/app/legacy/containers/Ad/MPU.jsx](https://github.com/bbc/simorgh/pull/10887/files#diff-bfc37a4782f6034694d35c08df892759c801ad802669f115ab57de07d094dfd6)

karinathomasbbc committed:
Use SLOT_TYPES consts instead of hardcoding

karinathomasbbc committed:
Update snapshots

karinathomasbbc committed:
Add typing to test and suppress partial data TS errors

karinathomasbbc committed:
Fix imports

karinathomasbbc committed:
Update canonical snapshots

karinathomasbbc committed:
Move MPU up a level

karinathomasbbc committed:
Fix imports for Ads

karinathomasbbc committed:
Use slot_types const

karinathomasbbc committed:
Update imports

karinathomasbbc committed:
Act is no longer required

karinathomasbbc committed:
Further tiy up

karinathomasbbc committed:
Merge branch 'latest' into migrate-ad-component-2

karinathomasbbc committed:
Merge branch 'migrate-ad-component-2' of github.com:bbc/simorgh into migrate-ad-component-2

karinathomasbbc committed:
Merge branch 'latest' of github.com:bbc/simorgh into migrate-ad-component-2

karinathomasbbc committed:
Fix merge conflict

karinathomasbbc committed:
Use Direction type for dir

karinathomasbbc committed:
Cast function to jest.Mock

karinathomasbbc committed:
Import Ad component from correct location

karinathomasbbc committed:
Remove invalid key prop from div

karinathomasbbc committed:
Remove StaticRouter from test

karinathomasbbc committed:
Use Direction type for dir

karinathomasbbc committed:
Revert "Remove invalid key prop from div"

This reverts commit 0a02c2875707fa32a7bb7c375af315895fbbf098.

karinathomasbbc committed:
Import Ad component from correct location & update tests

karinathomasbbc committed:
Fix MPU implementation + tests

karinathomasbbc committed:
Remove invalid key prop from div

karinathomasbbc committed:
Adding global props to allow ref attributes on divs

karinathomasbbc committed:
Update snapshot

karinathomasbbc committed:
Update import order

karinathomasbbc committed:
Update import order

karinathomasbbc committed:
Move amp access props for div to global def

karinathomasbbc committed:
Merge branch 'latest' of github.com:bbc/simorgh into migrate-ad-component-2

karinathomasbbc committed:
Merge branch 'latest' of github.com:bbc/simorgh into migrate-ad-component-2

karinathomasbbc committed:
Update snapshots in line with jest upgrade

karinathomasbbc committed:
Ignore amp-access and amp-access-hide props

karinathomasbbc committed:
Merge branch 'latest' into migrate-ad-component-2

karinathomasbbc commented on src/app/components/Ad/Amp/types.d.ts:
```diff
@@ -0,0 +1,26 @@
+   * As a workaround you can manually create custom types (https://stackoverflow.com/a/50601125).
+   */
+
+  interface AmpAd {
```
Should this perhaps be moved to the global.d.ts file?

amoore108 replied:
I think its ok to co-locate these to the AMP Ad component. We do that with the Image component for AMP Image.

karinathomasbbc commented on src/app/components/Ad/Amp/types.d.ts:
```diff
@@ -0,0 +1,26 @@
+
+  interface IntrinsicElements {
+    'amp-ad': AmpAd;
+    div: DivProps;
```
Is this needed since it is already declared in the global.d.ts file?

amoore108 replied:
Not entirely sure, can it be removed here without errors?

karinathomasbbc commented on src/app/components/Ad/Amp/AdSlot/index.test.tsx:
```diff
@@ -20,20 +20,29 @@ describe('getAssetType', () => {
 
 describe('getDataSlot', () => {
+  const originalAppEnv = process.env.SIMORGH_APP_ENV;
+
   afterEach(() => {
-    delete process.env.SIMORGH_APP_ENV;
+    process.env.SIMORGH_APP_ENV = originalAppEnv;
```
Should this have been left as it was  - i.e. keep the `delete process.env.SIMORGH_APP_ENV;` 🤔

karinathomasbbc committed:
Merge branch 'latest' into migrate-ad-component-2

karinathomasbbc committed:
Fix import order

karinathomasbbc committed:
Merge branch 'latest' of github.com:bbc/simorgh into migrate-ad-component-2

karinathomasbbc committed:
Merge branch 'latest' of github.com:bbc/simorgh into migrate-ad-component-2

karinathomasbbc committed:
TEMP: Legacy front pages on test environment for testing purposes

karinathomasbbc commented on src/app/routes/utils/regex/utils/index.js:
This is for testing purposes only - to make sure we can test services with ads https://latest--5d28eb3fe163f6002046d6fa.chromatic.com/?path=/story/docs-rendering-ads--page#services-with-ads-enabled

karinathomasbbc committed:
TEMP: homepages are only azeri,burmese and kyrgyz - all other services will be legacy front page for testing purposes

karinathomasbbc committed:
- Page types with ads do not need to be import / render the AdsBootstrapJS file - this is now taken care of by the Ads component itself

- Rename CanonicalAdsBootstrapJs to AdsBootstrapJs

- Simplify logic on Front page, Article, Story, Topic & Feature Index pages

karinathomasbbc committed:
Remove rogue ;

karinathomasbbc committed:
Move showAdsBasedOnLocation logic to the Ads component

karinathomasbbc committed:
Fix tests

karinathomasbbc committed:
Merge branch 'latest' of github.com:bbc/simorgh into migrate-ad-component-2

karinathomasbbc committed:
Remove div props as these are already declared at a higher level

karinathomasbbc committed:
Reinstate deletion of SIMORGH_APP_ENV environment variable

karinathomasbbc committed:
Delete unnecessary import

karinathomasbbc committed:
Update test

karinathomasbbc committed:
Simplify logic

karinathomasbbc committed:
Only display ads if allowAdvertising is true. The other criteria for whether ads should be displayed is contained within the Ad component

karinathomasbbc commented on src/app/routes/utils/regex/utils/index.js:
DO NOT MERGE!!

karinathomasbbc committed:
Revert "TEMP: homepages are only azeri,burmese and kyrgyz - all other services will be legacy front page for testing purposes"

This reverts commit c0d720b18fb1811b5aea871cddbf2b1a2a04429b.

karinathomasbbc committed:
Fix tests

karinathomasbbc committed:
Update comments

karinathomasbbc committed:
Remove invalid test

karinathomasbbc committed:
Revert "TEMP: Legacy front pages on test environment for testing purposes"

This reverts commit 9cc9c71a60d62f17d797859b932bdda3f077a735.

karinathomasbbc committed:
Increase test coverage for ad component

amoore108 commented on src/app/pages/StoryPage/StoryPage.jsx:
```diff
@@ -293,11 +290,9 @@ const StoryPage = ({ pageData }) => {
-      {isAdsEnabled && !isAmp && (
-        <CanonicalAdBootstrapJs adcampaign={adcampaign} />
+      {allowAdvertising && (
+        <AdContainer slotType="leaderboard" adcampaign={adcampaign} />
```
With the `AdContainer` including the `AdBootstrapJs`, will there be a problem rendering 2 of the bootstrap script tags if the MPU block and Leaderboard components are on the same page? Or maybe react-helmet is clever enough to prevent duplicate script tags?

karinathomasbbc replied:
That is a very good question (and tbh I never thought of that)! I will check and let you know.

karinathomasbbc replied:
I've checked on our preview environment, and I can only see one occurence of the script in the head of the html:

![image](https://github.com/bbc/simorgh/assets/58214768/b1f9bc35-9f8e-4de6-9ab2-902601cff148)

So I think(?) we should be good? 🤞

shayneahchoon reviewed (approved):
![image](https://media4.giphy.com/media/tIeCLkB8geYtW/giphy.gif?cid=ecf05e470ep160q5wk1bjuipoteadapv4ixmedtx6v7mziiu&ep=v1_gifs_search&rid=giphy.gif)

amoore108 reviewed (approved):
Great work here 🙏

karinathomasbbc committed:
Merge branch 'latest' into migrate-ad-component-2

karinathomasbbc committed:
Merge branch 'latest' of github.com:bbc/simorgh into migrate-ad-component-2

amoore108 committed:
Merge branch 'latest' into migrate-ad-component-2

karinathomasbbc committed:
Merge branch 'latest' into migrate-ad-component-2
